classes/quiz_settings.php

```diff
@@ -275,19 +275,54 @@ private function compute_config_key() {
      * @throws \coding_exception
      */
     private function compute_config() {
-        // Process all settings that are boolean.
-        $this->process_bool_settings();
+        switch ($this->get('requiresafeexambrowser')) {
+            case settings_provider::USE_SEB_NO:
+                break;
+            case settings_provider::USE_SEB_UPLOAD_CONFIG:
+                $this->process_seb_config_file();
+                break;
+            default:
+                // Process all settings that are boolean.
+                $this->process_bool_settings();
 
-        // Process quit settings.
-        $this->process_quit_settings();
+                // Process quit settings.
+                $this->process_quit_settings();
 
-        // Add all the URL filters.
-        $this->process_url_filters();
+                // Add all the URL filters.
+                $this->process_url_filters();
+        }
 
         // Export and save the config, ready for DB.
         $this->set('config', $this->plist->to_xml());
     }
 
+    /**
+     * If file is uploaded, save the file to the config field.
+     * This is processed after the validation step, so a SEB file should exist at this point.
+     *
+     * @throws \CFPropertyList\IOException
+     * @throws \CFPropertyList\PListException
+     * @throws \DOMException
+     * @throws \coding_exception
+     */
+    private function process_seb_config_file() {
+        $cm = get_coursemodule_from_instance('quiz', $this->get('quizid'));
+
+        $file = settings_provider::get_module_context_sebconfig_file($cm->id);
```

Comment:
we can probably get rid of using settings_provider here, if we would store id of the file as sebconfigfile. Thoughts?

Reply:
but then it will add complexity on backup and restore...

Reply:
Don't we still need the file somehow? Even if the file is referenced using the $cmid as the file $itemid, it must still be obtained from the file_storage api. This function returns the file itself to parse it and make changes to the quiz_settings plist.